test/aries-js-worker/test/messaging/messaging.js

```diff
@@ -120,6 +120,28 @@ describe("Basic Messaging", function () {
         assert.equal(incomingMsg["@id"], sampleReplyMsg["@id"])
         assert.equal(incomingMsg["@type"], sampleReplyMsg["@type"])
         assert.equal(incomingMsg.content, sampleReplyMsg.content)
+        assert.deepEqual(incomingMsg["~thread"], {
+            "thid": "1d9a1589-9d7b-4308-9fab-8ee9730720c2"
+        })
+    })
+
+    it("receiver replies to last received basic message by starting new thread", async function () {
+        receiver.messaging.reply({"message_ID": `${lastReceivedMsgID}`, "message_body": sampleReplyMsg, "start_new_thread": true})
+
+        const incomingMsg = await new Promise((resolve, reject) => {
+            const timer = setTimeout(_ => reject(new Error("time out waiting for incoming message")), 5000)
+            const stop = sender.startNotifier(msg => {
+                stop()
+                resolve(msg.payload.message)
+            }, [basicMsgSvcName])
+        })
+
+        assert.equal(incomingMsg["@id"], sampleReplyMsg["@id"])
+        assert.equal(incomingMsg["@type"], sampleReplyMsg["@type"])
+        assert.equal(incomingMsg.content, sampleReplyMsg.content)
+        assert.deepEqual(incomingMsg["~thread"], {
+            "pthid": "1d9a1589-9d7b-4308-9fab-8ee9730720c2"
```

Comment:
It would be better
```
// first startNotifier
const incoming = new Promise
// do action 
receiver.messaging.reply
// wait for startNotifier response 
incomingMsg await incoming
// checks
assert.equal(incomingMsg["@i

```

Reply:
I didn't get you, it is same variation of previous step. Can you elobarate?

Reply:
fixed